arrow-cpp is broken

It seems that all of a sudden a few dependencies are missing to `arrow-cpp`. As a start, I found the need to add `openssl`. The problem is with `libevent` which also makes the impression it's needed:

```
CMake Error at /nix/store/m33sn6h5kgfbk3z4avwmgxam1dny07i0-cmake-3.21.2/share/cmake-3.21/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
  By not providing "FindLibevent.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Libevent",
  but CMake did not find one.

  Could not find a package configuration file provided by "Libevent" with any
  of the following names:

    LibeventConfig.cmake
    libevent-config.cmake

  Add the installation prefix of "Libevent" to CMAKE_PREFIX_PATH or set
  "Libevent_DIR" to a directory containing one of the above files.  If
  "Libevent" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  /nix/store/pqw4n75kz23qgi11nims4l0yqvdkjlhg-thrift-0.15.0/lib/cmake/thrift/ThriftConfig.cmake:84 (find_dependency)
  cmake_modules/ThirdpartyToolchain.cmake:235 (find_package)
  cmake_modules/ThirdpartyToolchain.cmake:1311 (resolve_dependency)
  CMakeLists.txt:515 (include)
```

But our build of `libevent` doesn't provide cmake files. Building `libevent` with cmake instead of autoconf might help, but they don't use the cmake module `GNUInstallDirs` there which makes the transition harder.

cc @tobim @veprbl (maintainers) and @cpcloud .

--- a/pkgs/development/libraries/arrow-cpp/default.nix
+++ b/pkgs/development/libraries/arrow-cpp/default.nix
@@ -90,6 +90,7 @@ in stdenv.mkDerivation rec {
     "-DARROW_BUILD_TESTS=ON"
     "-DARROW_VERBOSE_THIRDPARTY_BUILD=ON"
     "-DARROW_DEPENDENCY_SOURCE=SYSTEM"
+    "-DThrift_SOURCE=AUTO" # search for Thrift using pkg-config (ThriftConfig.cmake requires OpenSSL and libevent)
     "-DARROW_DEPENDENCY_USE_SHARED=${if enableShared then "ON" else "OFF"}"
     "-DARROW_COMPUTE=ON"
     "-DARROW_CSV=ON"

